src/Castle.Core/project.json

```diff
@@ -21,14 +21,16 @@
     "System.Runtime": "4.0.20",
     "System.Runtime.Extensions": "4.0.10",
     "System.Threading": "4.0.10",
-    "System.Xml.XmlDocument": "4.0.0"
+    "System.Xml.XmlDocument": "4.0.0",
+    "System.Xml.XmlSerializer": "4.0.10"
   },
   "frameworks": {
     "dotnet": { }
   },
   "compilationOptions": {
     "warningsAsErrors": true,
-    "define": [ "FEATURE_NETCORE_REFLECTION_API" ]
+    "define": [ "FEATURE_NETCORE_REFLECTION_API", "FEATURE_NETCORE_EXCEPTION_API", "FEATURE_NETCORE_CONVERTER_API" ],
+    "keyFile": "../../buildscripts/CastleKey.snk"
```

Comment:
Why are the ASP.NET team strong name signing their test projects, why are they even strong naming their libraries. I thought strong name signing was only added into DNX for some annoying back compat reason for corefx assemblies.

Serilog doesn't strong name its .NET Core assemblies (.NET Framework ones are), which means the Castle ones can't be because `Castle.Services.Logging.SerilogIntegration` needs to reference it:
* https://github.com/serilog/serilog/blob/dev/src/Serilog/project.json#L60-L62
* https://github.com/nblumhardt/serilog/commit/4b8f81eb1744bb96a4ec3cf7150869809c1190f9

We don't use JSON.NET, however it isn't strong named:
* https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/project.json

We've all dealt with the pain of strong names and were basically promised that .NET Core would do away with all that mess. You have to provide a very compelling reason for us to want to do that again.